numpy/ma/core.py

```diff
@@ -198,14 +198,15 @@ class MaskError(MAError):
     default_filler["M8[" + v + "]"] = np.datetime64("NaT", v)
     default_filler["m8[" + v + "]"] = np.timedelta64("NaT", v)
 
+float_types_list = ['float16', 'float32', 'float64', 'float128',
+                    'complex64', 'complex128', 'complex256']
 max_filler = ntypes._minvals
-max_filler.update([(k, -np.inf) for k in [np.float32, np.float64]])
```

Comment:
Why can't you just add float16 to this line?

Reply:
The bug also affects complex values, so those needed to be included as well. At that point I thought it best to have a separate list of dtypes to loop over rather than putting them in individually

Reply:
I don't think the special case is needed at all if you use `longdouble` as a better spelling of `float128` - and similar spellings for all the other types (half, single, double, csingle, cdouble, clongdouble)

Reply:
One of the test cases failed at first when I called np.float128 directly, so I think the test may still be needed in some circumstances. I'll change the spellings to the ones you suggested though and see if that removes the need for the test

Reply:
Great, seems like changing the naming of the dtypes has removed the need for the checks

Reply:
There's no need to use typeDict any more either, np.single etc will work fine